### PR TITLE
Add test on fan / nested decoupled ensemble

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

File: qa/L0_decoupled/decoupled_test.py [hunk 1]
```diff
@@ -64,6 +66,8 @@ def setUp(self):
         self.outputs_ = []
         self.outputs_.append(grpcclient.InferRequestedOutput('OUT'))
         self.outputs_.append(grpcclient.InferRequestedOutput('IDX'))
+        # Some trials only expect a subset of outputs
+        self.requsted_outputs_ = self.outputs_
```
Comment (CoderHam) [inline qa/L0_decoupled/decoupled_test.py:70]: Typo -> requested_outputs_

File: qa/L0_decoupled/decoupled_test.py [hunk 2]
```diff
@@ -75,7 +79,7 @@ def _stream_infer(self, request_count, repeat_count, user_data, result_dict):
                 triton_client.async_stream_infer(model_name=self.model_name_,
                                                  inputs=self.inputs_,
                                                  request_id=str(i),
-                                                 outputs=self.outputs_)
+                                                 outputs=self.requsted_outputs_)
```
Comment (CoderHam) [inline qa/L0_decoupled/decoupled_test.py:82]: Typo -> requested_outputs_

File: qa/L0_decoupled/decoupled_test.py [hunk 3]
```diff
@@ -113,6 +138,9 @@ def _decoupled_infer(self,
         wait_data = np.array([wait_time], dtype=np.uint32)
         self.inputs_[2].set_data_from_numpy(wait_data)
 
+        # use validate_fn to differentiate requested outputs
+        self.requsted_outputs_ = self.outputs_ if validate_fn is None else self.outputs_[0:1]
```
Comment (CoderHam) [inline qa/L0_decoupled/decoupled_test.py:142]: Here too

File: qa/L0_model_config/autofill_noplatform/ensemble/invalid_decoupled_branching/expected [hunk 4]
```diff
@@ -0,0 +1 @@
+in ensemble invalid_decoupled_branching, step of model 'custom_nobatch_int32_int32_int32' receives inputs originated from different decoupled models
```
Comment (CoderHam) [inline qa/L0_model_config/autofill_noplatform/ensemble/invalid_decoupled_branching/expected:1]: add newline

File: qa/L0_model_config/autofill_noplatform/ensemble/invalid_decoupled_branching_2/expected [hunk 5]
```diff
@@ -0,0 +1 @@
+in ensemble invalid_decoupled_branching_2, step of model 'invalid_decoupled_branching_2' receives inputs originated from different decoupled models
```
Comment (CoderHam) [inline qa/L0_model_config/autofill_noplatform/ensemble/invalid_decoupled_branching_2/expected:1]: add newline

File: src/backends/backend/examples/identity.cc [hunk 6]
```diff
@@ -143,6 +143,7 @@ class ModelState {
   const std::string model_name_;
   const uint64_t model_version_;
   ni::TritonJson::Value model_config_;
+  bool support_batching_;
```
Comment (deadeyegoodwin) [inline src/backends/backend/examples/identity.cc:146]: Instead of these changes just need to get PR #1753 once it is merged.

**Comment by deadeyegoodwin** [comment]
@GuanLuo This needs to be manually rebased.

**Comment by deadeyegoodwin** [comment]
> @GuanLuo This needs to be manually rebased.

@GuanLuo Likely due to the python formatting. You probably just need to take format.py from master and use it to format decoupled_test.py.

**Comment by GuanLuo** [comment]
rebased

**Pull request merged**